app/main.py

```diff
@@ -1 +1,45 @@
-# write your code here
+class InvalidCommandFormat(Exception):
+    pass
+
+
+class InvalidCommand(Exception):
+    pass
+
+
+class SameFileError(Exception):
+    pass
+
+
+class SourceFileNotFoundError(Exception):
+    pass
+
+
+class PermissionDeniedError(Exception):
+    pass
+
+
+def copy_file(command: str) -> None:
+    commands = command.split()
+
+    if len(commands) != 3:
+        raise InvalidCommandFormat("Invalid command format. "
+                                   "Usage: cp source_file destination_file")
+
+    command_name, source_file, destination_file = commands
+
+    if command_name != "cp":
+        raise InvalidCommand("Invalid command. Use 'cp' to copy files.")
+
+    if source_file == destination_file:
```

Comment:
`It must do nothing in case the user is trying to copy file to file with the same name.`
this contradicts the task requirement

Reply:
it is better to use backward condition here, instead of blank return